packages/sync-actions/src/utils/extract-matching-pairs.js

```diff
@@ -0,0 +1,16 @@
+export default function extractMatchingPairs(hashMap, key, before, now) {
+  let oldObjPos
+  let newObjPos
+  let oldObj
+  let newObj
+
+  if (hashMap[key]) {
+    oldObjPos = hashMap[key][0]
+    newObjPos = hashMap[key][1]
+    if (before && before[oldObjPos]) oldObj = before[oldObjPos]
+
+    if (now && now[newObjPos]) newObj = now[newObjPos]
+  }
+
+  return { oldObj, newObj }
```

Comment:
There is of course a great opportunity to rename these to `previous` and `next`. I prefer that naming, tbh. Not sure tho if it is part of this PR, at least in the scope of a new utility.

Reply:
Nah I think that's something for another day. Maybe even across the whole library otherwise things might get even more confusing than clear :).